### ISSUE TITLE
Deck Editor makes the Application crash.

### ISSUE THREAD

**Issue by redhead4444** [Cockatrice/Cockatrice#4021]
<b>System Information:</b>
Client Version: 2.7.4 (2020-03-20)
Client Operating System: macOS High Sierra (10.13)
Build Architecture: 64-bit
Qt Version: 5.14.1
System Locale: es_ES
Install Mode: Standard

---------------------------------------------------------------------------

Theme changed: "Default"
SoundEngine: enabling sound
Sound theme changed: "Default"
SoundEngine: enabling sound
Language changed: ""
main(): starting main program
main(): MainWindow constructor finished
Spoilers Disabled
main(): ui.show() finished
Startup: found config with current version
CardDatabase::loadCardDatabases start
[CockatriceXml4Parser] Trying to parse:  "/Users/Antonio/Library/Application Support/Cockatrice/Cockatrice/cards.xml"
PictureLoader: cache fail for "_trice_card_back_232372"
[CardDatabase] loadCardDatabase(): Path = "/Users/Antonio/Library/Application Support/Cockatrice/Cockatrice/cards.xml" Status = 0 Cards = 20927 Sets= 483
[CockatriceXml4Parser] Trying to parse:  "/Users/Antonio/Library/Application Support/Cockatrice/Cockatrice/tokens.xml"
[CockatriceXml4Parser] Unknown item "info" , trying to continue anyway
[CardDatabase] loadCardDatabase(): Path = "/Users/Antonio/Library/Application Support/Cockatrice/Cockatrice/tokens.xml" Status = 0 Cards = 21390 Sets= 483
[CardDatabase] loadCardDatabase(): Path = "/Users/Antonio/Library/Application Support/Cockatrice/Cockatrice/spoiler.xml" Status = 4 Cards = 21390 Sets= 483
CardDatabase::loadCardDatabases success

_______________________________________________________________________________________

I'm building my MTG cube on Cockatrice (360 cards). If I make the list from scratch, cockatrice works perfectly, but once I save it and close the application, it won't load later on. I thought the cards were too many, so I made two new files, splitting the Cube in two. Again, when I create the "decks" they work just fine and the number of cards added don't seem to disturb the program. Next day, after closing the program and trying to load the decks, they actually load (the smaller ones, the whole cube still doesn't), but the program will crash everytime I try to add or replace more cards. The editor lets me delete cards from the decklist, but doesn't let me put any new card in, it just crashes the program. Unfortunately I'm super bad with computers and don't know anything about programming or coding, so I don't know how to solve this. Are the files just too big?

Running on:
MacBook Pro (13inch - Mid 2012)
2,5 GHz intel Core i5
16 GB 1600 MHz DDR3
Intel HD Graphics 4000 1536 MB

**Comment by tooomm** [comment]
Can you upload the problematic files somewhere and link them here, please?

Also, can you load the crashing deck (your cube) in a game without problems?

**Comment by ebbit1q** [comment]
For reference on max deck length, here is a deck with every single card currently available

![image](https://user-images.githubusercontent.com/36401181/83765654-bd93f500-a67b-11ea-99a6-fbb599a1139d.png)

It loads instantly, consider that it contains substantially less informatioin than the loading of the entire card database on startup.
I made it using a bash script if you'd like to make a deck like this with any database here it is:
```bash
cards="$HOME/.local/share/Cockatrice/Cockatrice/cards.xml"
first="<cards>"
pre='<?xml version="1.0" encoding="UTF-8"?>
<cockatrice_deck version="1">
    <deckname>big deck</deckname>
    <comments></comments>
    <zone name="main">
        '
end='</zone>
</cockatrice_deck>'
linepre='<card number="1" name="'
lineend='"/>
        '
len="$(grep -F "" -c "$cards")"
names="$(grep -F "$first" -A "$len" "$cards" | grep -Po '(?<=<name>).*(?=<\/name>)')"
echo "$pre$linepre${names//
/$lineend$linepre}$lineend$end"
```
It takes a long time to run :P

**Comment by redhead4444** [comment]
> Can you upload the problematic xml file of your cube somewhere and link them here, please?
 ---  I made a dropbox folder with all problematic files. Here it is:
https://www.dropbox.com/sh/0d58xio1xzmroiz/AABMxkGlpZcbFwAKCV13B2p0a?dl=0
unfortunately I don't know how to convert the .cod files into .xml. 

> Also, can you load the crashing deck (your cube) in a game without problems?
 --- Yes, the whole cube (the one that doesn't load at all in deck editor) loads in a local game. I        haven't tried in an online server, though!

Thank you very much for the quick answers!

**Comment by redhead4444** [comment]
> For reference on max deck length, here is a deck with every single card currently available
> 
> ![image](https://user-images.githubusercontent.com/36401181/83765654-bd93f500-a67b-11ea-99a6-fbb599a1139d.png)
> 
> It loads instantly, consider that it contains substantially less informatioin than the loading of the entire card database on startup.
> I made it using a bash script if you'd like to make a deck like this with any database here it is:
> 
> ```shell
> cards="$HOME/.local/share/Cockatrice/Cockatrice/cards.xml"
> first="<cards>"
> pre='<?xml version="1.0" encoding="UTF-8"?>
> <cockatrice_deck version="1">
>     <deckname>big deck</deckname>
>     <comments></comments>
>     <zone name="main">
>         '
> end='</zone>
> </cockatrice_deck>'
> linepre='<card number="1" name="'
> lineend='"/>
>         '
> len="$(grep -F "" -c "$cards")"
> names="$(grep -F "$first" -A "$len" "$cards" | grep -Po '(?<=<name>).*(?=<\/name>)')"
> echo "$pre$linepre${names//
> /$lineend$linepre}$lineend$end"
> ```
> 
> It takes a long time to run :P

Hey, thank you for the recommendation. Unfortunately I don't know what a "bash script" is nor how to use it. I just wanted to report this here, because I respect and value a lot Cockatrice as an Iniciative and as a support for the TCGs community (mainly for MTG players). I have no real interest in learning how to use code, etc. in the computer though, tbh... But again, thanks a lot for your efforts! :)

**Comment by tooomm** [comment]
@redhead4444 I can load all of your provided .cod decks from the Dropbox just fine.
But I'm running a newer client version. Maybe you want to try a more recent beta release?

But I have to say, that only `Cube (1)` is a viable deck. The other two files are messed up and just contain very strange content - they load just fine however:
![Untitled](https://user-images.githubusercontent.com/9874850/83849827-4d837e80-a710-11ea-83be-edbcba72c37f.png)

**Comment by ebbit1q** [comment]
@tooomm none of these uploaded files contain that, I think you failed to download the correct file, this looks like a webpage in fact.

**Comment by ebbit1q** [comment]
@redhead4444 all three these files open up without problem on your version, development versions and most likely any previous version of cockatrice to feature the .cod format (which is implemented in xml)
![image](https://user-images.githubusercontent.com/36401181/83883916-7160b780-a744-11ea-9bd4-560ad6819b95.png)

The only remaining variable here is your specific system and setup, maybe uninstalling and reinstalling cockatrice would sort this out (make sure to save your decks somewhere safe)

**Comment by tooomm** [comment]
>@tooomm none of these uploaded files contain that, I think you failed to download the correct file, this looks like a webpage in fact.

I wondered already... 😆 
Redownloading gives me the right files. They work for me, too @redhead4444.

**Comment by Olxinos** [comment]
Hi, I'm 3 months late, but I suspect this was the same bug as #3834 and #3903 that should have been fixed in 2.7.5 (by #3967 ). The symptoms are similar (crash on load on OSX), I could reproduce the crash with these files on an old 2.7.4 version that was lying around (on OSX 10.13), but the 2.7.5 version could load and edit these decklists perfectly fine (same computer).

Anyways, it's probably already fixed in the last version (although we'd need @redhead4444 to check again with the 2.7.5 or later to be absolutely sure)

**Comment by ebbit1q** [comment]
Yeah, it looks exactly the same, closing for now.
If you're still experiencing this issue please reopen it with a new system info excerpt from the log.
